Add blog post: Automatic Issue Detection (#557)
mlflow/mlflow-website: website/blog/2026-04-09-issue-detection/index.md

---

> title: "Tired of Manually Reviewing Traces to Find Issues in Your Agent? Meet Automatic Issue Detection"

Reviewer: ```suggestion
title: "Tired of Reviewing Traces? Meet Automatic Issue Detection for Your Agent"
```

Love this framing! Minor suggestion to make the title a bit shorter.

tags:
  [genai, evaluation, quality, ai-insights, tracing, observability, automatic]
slug: issue-detection
authors: [mlflow-maintainers]
thumbnail: /img/blog/issue-detection-thumbnail.png
image: /img/blog/issue-detection-thumbnail.png
---

> Detecting issues in your agent is tedious, inconsistent, and hard to scale. Today we're announcing [**Automatic Issue Detection**](https://mlflow.org/docs/latest/genai/eval-monitor/ai-insights/detect-issues/), a new MLflow AI Insights feature that makes it simpler to find and triage problems directly from your LLM application traces.

Reviewer: ```suggestion
[Observability](https://mlflow.org/docs/latest/genai/tracing/) has became a norm for AI agents in production. has become a norm for AI agents in production. But recording logs, metrics, and traces alone doesn't make the user experience better. You need to act on the data.
 
The problem is that finding actionable insights from massive traces is a needle-in-a-haystack problem. From thousands of production logs, how do you spot conversations where users got frustrated or dropped out? How do you catch an agent that silently returns wrong answers with high confidence and misleads users? Manually reviewing traces one by one doesn't scale.

Today we're announcing [**Automatic Issue Detection**](https://mlflow.org/docs/latest/genai/eval-monitor/ai-insights/detect-issues/) in MLflow, a new AI-driven Insights feature that replaces hours of manual check and triage to just 3 clicks.
```

nit: I'd suggest adding a user-oriented problem statement like this at the beginning. This is a new category of solution so we need to convince ppl this is A problem to pay attention. We have "Why Do You Need Automatic Issue Detection?" section but it is details in one further layer.

<video width="100%" controls autoPlay loop muted>
  <source
    src={require("@site/static/img/releases/3.11.1/issue-detection.mp4").default}
    type="video/mp4"
  />
</video>

## Why Do You Need Automatic Issue Detection?

As LLM applications grow in production, maintaining agent quality becomes increasingly challenging. Four problems compound as traffic scales:

- **Manual review doesn't scale:** Inspecting individual traces one by one is unsustainable as request volume grows.

> - **Unclear metrics:** It's hard to know which quality dimensions to measure without predefined baselines to start from.

Reviewer: ```suggestion
- **Unclear criteria:** It's hard to know which quality dimensions to measure without predefined baselines to start from.
```

I think you are trying to say the criteria of measuring quality is vague right? "Unclear metrics" can sound like the metric data is unclear instead of definition.

- **Scattered failure patterns:** Related failures spread across thousands of traces with no systematic grouping, making recurring issues easy to miss.
- **Unstructured tracking:** Without formal issue management, identified problems disappear into notes and Slack threads—and regressions go undetected.

Automatic Issue Detection moves teams from reactive, manual debugging to proactive, systematic quality identification.

## How It Works

> Issue Detection is built into the MLflow UI and runs directly against the traces you've already collected.
> 
> ![EntryPoint](./detect-issues-button.png)

Reviewer: ```suggestion
## Getting Started
Issue Detection is built into the MLflow UI and runs directly against the traces you've already collected.

![EntryPoint](./detect-issues-button.png)

## How It Works

### The CLEARS Framework
```

nit: "How It Works" is mainly about internal methodology. The screenshot explains the button is more like how to use the feature.

### The CLEARS Framework

> MLflow organizes issue detection across six quality dimensions, forming the **CLEARS** framework (**C**orrectness, **L**atency, **E**xecution, **A**dherence, **R**elevance, **S**afety). Choose which categories to focus on based on your application's requirements:

Reviewer: Can we add link to the blog where we explain CELARS framework?
(Also we should add backlink from there "here is how you can automatically detect these all at once ->")

Author: The CLEARS blog is not published yet, i'll link to https://mlflow.org/docs/latest/genai/eval-monitor/ai-insights/detect-issues/#issue-categories-clears for now

![Categories](./categories.png)

You choose which CLEARS categories matter for your use case. A customer support bot might prioritize Adherence and Safety; a code assistant cares most about Correctness and Execution.

### The Detection Pipeline

Once you start an analysis run, MLflow:

1. **Samples and analyzes** traces using an LLM of your choice (via MLflow AI Gateway or a direct API connection)
2. **Clusters** related problems so you see patterns, not just a flat list of individual failures
3. **Annotates** the source traces with specific findings so you can drill straight to the evidence
4. **Generates a summary** of key findings, severity distribution, and recommended next steps

The analysis runs asynchronously with real-time progress tracking, so you can kick it off and come back to the results.

![In-progress Job](./issue-detection-progress.png)

> ![Complete Job](./issue-detection-complete.png)

Reviewer: nit: The split between two images is invisible so it looks like a messy single page. Perhaps we can drop the second image? I feel the `issue-tracking.png` visualize how the detected issue looks lik pretty well.
